canister/src/api/get_block_headers.rs

```diff
@@ -1,17 +1,139 @@
+use bitcoin::consensus::Encodable;
 use ic_btc_interface::{GetBlockHeadersError, GetBlockHeadersRequest, GetBlockHeadersResponse};
 
-fn verify_get_block_headers_request(
-    request: GetBlockHeadersRequest,
-) -> Result<(), GetBlockHeadersError> {
+use crate::{
+    charge_cycles,
+    runtime::{performance_counter, print},
+    state::main_chain_height,
+    verify_has_enough_cycles, with_state, with_state_mut,
+};
+
+// Various profiling stats for tracking the performance of `get_block_headers`.
+#[derive(Default, Debug)]
+struct Stats {
+    // The total number of instructions used to process the request.
+    ins_total: u64,
+
+    // The number of instructions used to build the block headers vec.
+    ins_build_block_headers_vec: u64,
+}
+
+fn verify_requested_height_range_and_return_effective_range(
+    request: &GetBlockHeadersRequest,
+) -> Result<(u32, u32), GetBlockHeadersError> {
```

Comment:
What happens if a user requests 100k headers? We need to bound how many headers are returned, otherwise we'll run out of instructions. I suggest bounding it to 100 for now. @THLO WDYT?

Reply:
Done.

Reply:
PTAL.